packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -370,16 +375,36 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
         commands.registerCommand({ id: 'workbench.files.action.refreshFilesExplorer' }, {
             execute: () => commands.executeCommand(FileNavigatorCommands.REFRESH_NAVIGATOR.id)
         });
-        commands.registerCommand({ id: VscodeCommands.INSTALL_FROM_VSIX.id }, {
+        commands.registerCommand(VscodeCommands.INSTALL_FROM_VSIX, {
             execute: async (vsixUriOrExtensionId: TheiaURI | UriComponents | string) => {
                 if (typeof vsixUriOrExtensionId === 'string') {
-                    await this.pluginServer.deploy(VSCodeExtensionUri.fromId(vsixUriOrExtensionId).toString());
+                    let extensionId = vsixUriOrExtensionId;
+                    let opts: PluginDeployOptions | undefined;
+                    if (PluginIdentifiers.isVersionedId(vsixUriOrExtensionId)) {
+                        const idAndVersion = PluginIdentifiers.getIdAndVersion(vsixUriOrExtensionId);
+                        extensionId = idAndVersion[0];
+                        opts = { version: idAndVersion[1]!, ignoreOtherVersions: true };
+                    }
+                    await this.pluginServer.deploy(VSCodeExtensionUri.fromId(extensionId).toString(), undefined, opts);
                 } else {
                     const uriPath = isUriComponents(vsixUriOrExtensionId) ? URI.revive(vsixUriOrExtensionId).fsPath : await this.fileService.fsPath(vsixUriOrExtensionId);
                     await this.pluginServer.deploy(`local-file:${uriPath}`);
                 }
             }
         });
+        commands.registerCommand(VscodeCommands.UNINSTALL_EXTENSION, {
+            execute: async (id: string) => {
+                if (!id) {
+                    throw new Error(nls.localizeByDefault('Extension id required.'));
+                }
+                if (!PluginIdentifiers.isVersionedId(id)) {
```

Comment:
I don't think this makes sense from a user's perspective: while technically, we can have multiple versions of the same plugin installed, that's not how it's presented to the user: for them, you either have a plugin installed, or not, independent of the version, so we should probably pass a naked id here.

Reply:
`PluginServer.uninstall()` expects `PluginIdentifiers.VersionedId` parameter which is typed as `${string}.${string}@${string}` that why it the code above makes sense to me. 

> that's not how it's presented to the user: for them, you either have a plugin installed, or not

The `UNINSTALL_EXTENSION` command is not presented to the user, but is used in `theia/packages/vsx-registry/src/browser/vsx-extension.tsx` where the versioned extension id is given. 
I would prefer to keep require versioned id here.